### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by nbhd**
https://github.com/nbhd/welcome-to-android/issues/5
- [x] @eoinfogarty 
- [x] @bowyer-app

### CODE REVIEW

File: ActivitiesAndIntents/app/src/main/java/net/boondockradio/activitiesandintents/MainActivity.java [hunk 1]
```diff
@@ -12,34 +15,38 @@ protected void onCreate(Bundle savedInstanceState) {
...
-
-            case R.id.btn_activity3:
-                ThreeActivity.startActivity(this);
-                break;
-
-            case R.id.btn_activity4:
-                FourActivity.startActivity(this);
-                break;
-        }
+        EditText edit = (EditText) findViewById(R.id.edit_txt);
```
Comment (eoinfogarty) [inline ActivitiesAndIntents/app/src/main/java/net/boondockradio/activitiesandintents/MainActivity.java]: you dont need to do this if you make `edit` a member variable. i.e.

``` java

public class MainActivity extends AppCompatActivity implements View.OnClickListener {

     private EditText mEditText;

     @Override
     protected void onCreate(Bundle savedInstanceState) {
          super.onCreate(savedInstanceState);
          setContentView(R.layout.activity_main);

          mEditText = (EditText) findViewById(R.id.edit_txt);
          ...
    }

     public void onClick(View v) {
           String message = mEditText.getText().toString();
           DetailActivity.startActivity(this, message);
     }
}
```

File: ActivitiesAndIntents/app/src/main/res/layout/activity_detail.xml [hunk 2]
```diff
@@ -7,13 +7,13 @@
     android:paddingLeft="@dimen/activity_horizontal_margin"
     android:paddingRight="@dimen/activity_horizontal_margin"
     android:paddingTop="@dimen/activity_vertical_margin"
-    tools:context="net.boondockradio.activitiesandintents.OneActivity">
+    tools:context=".DetailActivity">
 
     <TextView
         android:id="@+id/txt_activity_one"
```
Comment (eoinfogarty) [inline ActivitiesAndIntents/app/src/main/res/layout/activity_detail.xml]: we should try keep naming consistant

**Comment by shoheikawano** [comment]
So many deletions! Amazing! 😄

File: ActivitiesAndIntents/app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -13,34 +13,21 @@
...
-        android:layout_height="wrap_content"
-        android:layout_centerInParent="true"
-        android:layout_below="@+id/btn_activity1"
-        android:text="@string/label_activity2"/>
-    <Button
-        android:id="@+id/btn_activity3"
-        android:layout_width="wrap_content"
+
+    <EditText
+        android:id="@+id/edit_txt"
```
Comment (eoinfogarty) [inline ActivitiesAndIntents/app/src/main/res/layout/activity_main.xml]: `android:id="@+id/edit_activity_main"`

File: ActivitiesAndIntents/app/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -13,34 +13,21 @@
...
         android:layout_height="wrap_content"
+        android:layout_above="@+id/btn_activity1"
         android:layout_centerInParent="true"
-        android:layout_below="@+id/btn_activity2"
-        android:text="@string/label_activity3"/>
+        android:layout_marginBottom="@dimen/main_activity_edit_text_margin"/>
+
     <Button
-        android:id="@+id/btn_activity4"
+        android:id="@+id/btn_activity1"
```
Comment (eoinfogarty) [inline ActivitiesAndIntents/app/src/main/res/layout/activity_main.xml]: `android:id="@+id/btn_activity_main"`

File: ActivitiesAndIntents/app/src/main/java/net/boondockradio/activitiesandintents/MainActivity.java [hunk 5]
```diff
@@ -12,34 +15,38 @@ protected void onCreate(Bundle savedInstanceState) {
...
+            }
+
+            @Override
+            public void onTextChanged(CharSequence s, int start, int before, int count) {
+                if (s.length() == 0) {
+                    button.setEnabled(false);
+                    return;
+                }
+
+                button.setEnabled(true);
```
Comment (eoinfogarty) [inline ActivitiesAndIntents/app/src/main/java/net/boondockradio/activitiesandintents/MainActivity.java]: this is ok, can also be written as `button.setEnabled(s.length() != 0);` i think

Comment (eoinfogarty) [inline ActivitiesAndIntents/app/src/main/java/net/boondockradio/activitiesandintents/MainActivity.java]: or for better readability

``` java
boolean hasText = s.length() != 0;
button.setEnabled(hasText);
```

Comment (nbhd) [inline ActivitiesAndIntents/app/src/main/java/net/boondockradio/activitiesandintents/MainActivity.java]: it is simple!
thx

**Comment by eoinfogarty** [comment]
LGTM

**Comment by bowyer-app** [comment]
LGTM

**Pull request merged**